test/cmdlineTests.sh

```diff
@@ -317,47 +317,56 @@ function test_via_ir_equivalence()
         fail "The second argument must be --optimize if present."
     fi
 
-    local solidity_code="$1"
+    local solidity_file="$1"
     local optimize_flag="$2"
 
+    filename=$(basename "$1" .sol)
+
     local optimizer_flags=()
     [[ $optimize_flag == "" ]] || optimizer_flags+=("$optimize_flag")
+    [[ $optimize_flag == "" ]] || filename+="_optimize"
 
-    local ir_output
-    ir_output=$(
-        echo "$solidity_code" |
-        msg_on_error --no-stderr "$SOLC" - --ir-optimized --debug-info location "${optimizer_flags[@]}" |
-        sed '/^Optimized IR:$/d'
-    )
+    msg_on_error --no-stderr "$SOLC" --ir-optimized --debug-info location "${optimizer_flags[@]}" "$solidity_file" |
+    sed '/^Optimized IR:$/d' |
+    awk -v RS= "{print > (\"${filename}_\"NR \".yul\")}"
 
-    local asm_output_two_stage asm_output_via_ir
-    asm_output_two_stage=$(
-        echo "$ir_output" |
-        msg_on_error --no-stderr "$SOLC" - --strict-assembly --asm "${optimizer_flags[@]}" |
+    for yul_file in $(find . -name "$filename*.yul" | sort -V); do
+        msg_on_error --no-stderr "$SOLC" --strict-assembly --asm "${optimizer_flags[@]}" "$yul_file" |
         sed '/^======= <stdin>/d' |
-        sed '/^Text representation:$/d'
-    )
+        sed '/^Text representation:$/d' >> "${yul_file/.yul/.asm}"
+    done
+
+    local asm_output_two_stage asm_output_via_ir
+    for asm_file in $(find . -name "$filename*.asm" | sort -V); do
+        asm_output_two_stage+=$(sed '/^asm_output_two_stage:$/d' "$asm_file" | sed '/^=======/d')
+    done
+
     asm_output_via_ir=$(
-        echo "$solidity_code" |
-        msg_on_error --no-stderr "$SOLC" - --via-ir --asm --debug-info location "${optimizer_flags[@]}" |
+        msg_on_error --no-stderr "$SOLC" --via-ir --asm --debug-info location "${optimizer_flags[@]}" "$solidity_file" |
         sed '/^======= <stdin>/d' |
-        sed '/^EVM assembly:$/d'
+        sed '/^EVM assembly:$/d' |
+        sed '/^asm_output_via_ir/d' |
```

Comment:
Not sure what this line is meant to do. `asm_output_via_ir` is a variable but here you're expecting to find its name in compiler output. It would make more sense if you just forgot `$` but then it wouldn't work either because it's in single quotes, because it's not yet defined at this point and because this sed command requires a single-line string.

Same with `asm_output_two_stage` in the loop.